app/services/new_version_service.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# Duplicates a version, creating new related objects rather than point at objects for a past version.
+# This is like NewVersionParameter, but it works on objects instead of parameters.
+class NewVersionService
```

Comment:
There is no need to merge this PR.  We can just do the migration on this branch.